File: 0974-subarray-sums-divisible-by-k/0974-subarray-sums-divisible-by-k.cpp
class Solution {
public:
    int subarraysDivByK(vector<int>& nums, int k) {
        vector<int> freqCount(k);
        freqCount[0] = 1;
        int prefix = 0, res = 0;
        for (int a : nums) {
            prefix = (prefix + a % k + k) % k;
            res += freqCount[prefix]++;
        }
        return res;
    }
};

File: 0974-subarray-sums-divisible-by-k/0974-subarray-sums-divisible-by-k.java
class Solution {
    public int subarraysDivByK(int[] nums, int k) {
        int[] freqCount = new int[k];
        freqCount[0] = 1;
        int prefix = 0, result = 0;
        for (int num : nums) {
            prefix = (prefix + num % k + k) % k;
            result += freqCount[prefix];
            freqCount[prefix] += 1;
        }
        return result;
    }
}